doc/modules/impute.rst

```diff
@@ -180,6 +187,42 @@ References
 .. [2] Roderick J A Little and Donald B Rubin (1986). "Statistical Analysis
    with Missing Data". John Wiley & Sons, Inc., New York, NY, USA.
 
+.. _knnimpute:
+
+Nearest neighbors imputation
+============================
+
+The :class:`KNNImputer` class provides imputation for filling in missing
+values using the k-Nearest Neighbors approach. By default, a eucliean distance
+metric that supports missing values,
+:func:`~sklearn.metrics.nan_euclidean_distances`, is used to find the
+nearest neighbors. Each missing feature are 
+imputed using values from ``n_neighbors`` nearest neighbors that have a value
+for the feature. The feature of the neighbors are averaged uniformly or
+weighted by distance. When the number of neighbors is less than 
+``n_neighbors``, the training set average for that feature is used for
```

Comment:
This is not really a special case if the regression is unweighted. If it's weighted, I'm not sure why the mean of all training values is a good choice.

Reply:
For the weighted case, we can weight this mean by distances as well.

Reply:
For the unweighted case: I might have argued this point wrongly before? It could be that the feature is present in a lot of samples, but none of them have features in common with the current sample, so they are not considered neighbors. So the mean includes samples that are not neighbors.

For the weighted case: I still think it's reasonable there.